File: BoardGameUnity/Assets/Scripts/GameBrowser/DBAccessor.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

namespace GameBrowser {
    // all query result is returned in string format
    // usually, they are the json format of specifc format
    public class DBAccessor : MonoBehaviour {
     
        public delegate void OnQueryResultBack(string queryResult);

        private int queryIDCounter = 0;
        // key: query id, value: callback
        private Dictionary<int, OnQueryResultBack> queriesWaitingForResponse = new Dictionary<int, OnQueryResultBack>();
        
        public void ProcessDBQueryResponse(ResponseMessage response) {
            Debug.Log("[DB Query]-response: " + response.content);
            var queryResponse = (DBQueryResponse)FullSerializerWrapper.Deserialize(typeof(DBQueryResponse), response.content);
            queriesWaitingForResponse[queryResponse.queryID](queryResponse.queryResult);
        }

        //  qurey result:
        //  json format "{string[]: cardnames}"
        public void GetRegisteredCardnames(OnQueryResultBack onQueryResultBack) {
            RegisterQuery("registeredCardnames", onQueryResultBack);
        }

        //  qurey result:
        //  json format {string[]: buffnames}
        public void GetRegisteredBuffnames(OnQueryResultBack onQueryResultBack) {
            RegisterQuery("registeredBuffnames",onQueryResultBack);
        }

        private void RegisterQuery(string querySentence, OnQueryResultBack onQueryResultBack) {
            var query = new DBQuery(queryIDCounter++, querySentence);
            GameBrowser.Instance.userInputManager.RegisterDBQuery(query);
            queriesWaitingForResponse[query.queryID] = onQueryResultBack;
        }
    }
}

File: BoardEngine/executables/pyproj/backend/db_accessor.py
from .protocol import DBQuery
from db.game_database import GameDatabase
import json

class DBAccessor:
    def __init__(self,db:GameDatabase):
        self.db = db

    def process_dbquery(self,dbquery:DBQuery)->str:
        result = {}
        if dbquery.query_sentence == 'registeredCardnames':
            cardnames =[cardname for cardname in self.db.game_app_data.cards_dict.keys()] 
            result = {"cardnames":cardnames}
        elif dbquery.query_sentence == 'registeredBuffnames':
            buffnames =[buffname for buffname in self.db.game_app_data.registered_buffnames] 
            result = {"buffnames":buffnames}
        return json.dumps(result)